src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -636,6 +645,121 @@ internal CompletionItem ResolveCompletionItem(CompletionItem completionItem)
+            string tokenText = token.Text.Replace("]",String.Empty).Replace("[",String.Empty);
+
+
+            if (scriptParseInfo.IsConnected && Monitor.TryEnter(scriptParseInfo.BuildingMetadataLock))
+            {
+                try
+                {
+                    // Queue the task with the binding queue    
+                    QueueItem queueItem = this.BindingQueue.QueueBindingOperation(
+                        key: scriptParseInfo.ConnectionKey,
+                        bindingTimeout: LanguageService.PeekDefinitionTimeout,
+                        bindOperation: (bindingContext, cancelToken) =>
+                        {
+                            // Get suggestions for the token
+                            IEnumerable<Declaration> declarationItems;
+                            int parserLine = textDocumentPosition.Position.Line + 1;
+                            int parserColumn = textDocumentPosition.Position.Character + 1;
+                            declarationItems = Resolver.FindCompletions(
+                                scriptParseInfo.ParseResult, 
+                                parserLine, parserColumn, 
+                                bindingContext.MetadataDisplayInfoProvider);
+
+                            // Match token with the suggestions(declaration items) returned
+                            DeclarationType type = 0;
+                            string schemaName;
+                            PeekDefinition peekDefinition = new PeekDefinition(connInfo);
+                            foreach (Declaration declarationItem in declarationItems)
+                            {
+                                if (declarationItem.Title.Equals(tokenText))
+                                {
+                                    type = declarationItem.Type;
+                                    // Script object using SMO based on type
+                                    switch (type)
+                                    {
+                                        case DeclarationType.Table:
+                                            return peekDefinition.GetTableDefinition(tokenText);
+                                        case DeclarationType.View:
+                                            schemaName = this.GetSchemaName(scriptParseInfo, textDocumentPosition.Position, scriptFile);
+                                            return peekDefinition.GetViewDefinition(tokenText, schemaName);
+                                        case DeclarationType.StoredProcedure:
+                                            schemaName = this.GetSchemaName(scriptParseInfo, textDocumentPosition.Position, scriptFile);
+                                            return peekDefinition.GetStoredProcedureDefinition(tokenText, schemaName);
+                                        default:
+                                            return null;
+                                    }
```

Comment:
This logic should be inside the PeekDefinition class. This service shouldn't know about these implementation details. Given you return null if not found, cleanest thing is move this info the peekDefinition class as-is for now and just call into it.
Separately, we're going to run into scale issues here - there are lots of types in SQL, so we shouldn't rely on switch statement as we'll have 100 entries before long. Options to replace:
- Use a Dictionary<Type,Func<string,string, Location[]> mapping type to a method that supports getting a definition.
- plenty of other patterns, but I'd guess this will work well enough for now. In the long term it might be nice to use dependency injection and have classes that register with supported types so the main logic class never needs to know about this, but it's overkill for now.